src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/DateTimeMarshaler.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices.JavaScript
+{
+    public static class DateTimeMarshaler
+    {
+        public static string FromJavaScriptPreFilter => "return BINDING._pre_filter_date(value)";
+        public static string ToJavaScriptPostFilter => "return new Date(value)";
+
+        public static DateTime FromJavaScript (double msecsSinceEpoch)
+        {
+            return DateTimeOffset.FromUnixTimeMilliseconds((long)msecsSinceEpoch).UtcDateTime;
+        }
+
+        public static double ToJavaScript (in DateTime dt)
+        {
+            return (double)((new DateTimeOffset(dt)).ToUnixTimeMilliseconds());
```

Comment:
All JS numbers are doubles, so this preserves the double input / double output relationship

Reply:
This will be explained clearly in documentation, but I don't think an explicit comment in the C# is necessary here to explain why it's a double since it's a basic JS thing. Feel free to unresolve this conversation if you really want me to add a comment anyway